### DISCUSSION TITLE
About IDirect3DDevice

### DISCUSSION THREAD

**Discussion by Lookforworld** [pywinrt/pywinrt#17] [Q&A]
@dlech 
When I use Direct3D11CaptureFramePool.Create() function,it needs a IDirect3DDevice to create,I read the [document](https://learn.microsoft.com/en-us/uwp/api/windows.graphics.directx.direct3d11.idirect3ddevice?view=winrt-22621), but I still don't know how to create it in python,I can't find a way to create it in our library,is there any way to do that?
Thank you very much！

**Comment by dlech** [comment]
Please see https://github.com/pywinrt/python-winsdk/issues/11

**Comment by Lookforworld** [comment]
> Please see [pywinrt/python-winsdk#11](https://github.com/pywinrt/python-winsdk/issues/11)

wow,it's very helpful！Thank you very much for your help！
Is the winsdk the same as winrt? Can they be replaced equivalently?

**Comment by dlech** [comment]
Yes.

**Comment by Lookforworld** [comment]
@dlech 
I checked winrt and winsdk and found no part about the windows.ui.composition,is there any way to achieve it?

**Comment by dlech** [comment]
I don't know enough about how composition works to know if it can be implemented or not.

**Comment by Lookforworld** [comment]
@dlech This stuff is really complicated！I found the content [here](https://github.com/ynkdir/py-win32more/tree/c9d52b917b12e5b5cdcc403458e8d7094233b5ef/win32more/System/WinRT), but I haven't been able to test it yet
